core/src/core_tests/workflow_tasks.rs

```diff
@@ -1694,3 +1695,48 @@ async fn tasks_from_completion_are_delivered() {
     .unwrap();
     core.shutdown().await;
 }
+
+#[tokio::test]
+async fn evict_missing_wf_during_poll_doesnt_eat_permit() {
+    test_telem_console();
+
+    let wfid = "fake_wf_id";
+    let mut t = TestHistoryBuilder::default();
+    t.add_by_type(EventType::WorkflowExecutionStarted);
+    t.add_full_wf_task();
+    t.add_we_signaled("sig", vec![]);
+    t.add_full_wf_task();
+    t.add_workflow_execution_completed();
+
+    let tasks = [hist_to_poll_resp(
+        &t,
+        wfid.to_owned(),
+        // Use a partial task so that we'll fetch history
+        ResponseType::OneTask(2),
+        TEST_Q.to_string(),
+    )];
+    let mut mock = mock_workflow_client();
+    mock.expect_get_workflow_execution_history()
```

Comment:
This is for paginating when workflow run is not in cache, right?

Reply:
Yup